packages/frontend/editor-ui/src/components/ParameterInput.vue

```diff
@@ -259,10 +259,12 @@ const editorIsReadOnly = computed<boolean>(() => {
 	return getTypeOption<boolean>('editorIsReadOnly') ?? false;
 });
 
-const editorLanguage = computed<CodeNodeEditorLanguage>(() => {
-	if (editorType.value === 'json' || props.parameter.type === 'json')
-		return 'json' as CodeNodeEditorLanguage;
-	return getTypeOption<CodeNodeEditorLanguage>('editorLanguage') ?? 'javaScript';
+const editorLanguage = computed<CodeNodeLanguageOption>(() => {
+	if (editorType.value === 'json' || props.parameter.type === 'json') return 'json';
+
+	if (node.value?.parameters?.language === 'pythonNative') return 'pythonNative';
+
+	return getTypeOption<CodeNodeLanguageOption>('editorLanguage') ?? 'javaScript';
```

Comment:
Not familiar with this code, but isn't the `pythonNative` another option in the node? So shouldn't this return it correctly?

Reply:
Not super familiar either but from a quick look it's inferring the language from `parameter.typeOptions.editorLanguage` in the node definition, rather than from the specific `language` param in the Code node, possibly because this setup needs to work with editors in params outside the Code node as well.